src/sys/socket/ffi.rs

```diff
@@ -37,7 +37,7 @@ pub struct cmsghdr {
     pub cmsg_len: type_of_cmsg_len,
     pub cmsg_level: c_int,
     pub cmsg_type: c_int,
-    pub cmsg_data: [usize; 0]
+    pub cmsg_data: [size_t; 0]
```

Comment:
Is this correct for all platforms? Or similarly, was using `type_of_cmsg_len` _totally_ wrong?